### ISSUE TITLE
cli: brew install pre versions

### ISSUE THREAD

**Issue by matti** [kontena/kontena#3070]
omnibus installer bundles with an ancient ruby, brew has 2.4.2

**Comment by kke** [comment]
You can `brew install --HEAD kontena` and you get the version from current master branch.

**Comment by matti** [comment]
tried that yesterday - it's difficult to tell from the --version of that if it's master or not (version says 1.4.0) - would be better to have the tagged pre version when installed by multiple developers and used in scripts.

**Comment by kke** [comment]
Yes the current master does not have the version and changelog that are in the v1.4 branch, but has all the closed PR's.

The version string has the commit hash, but it's a lot of work to figure out what that checkpoint includes.

**Comment by matti** [comment]
yep - it's better to have all developers sharing the same exact version than install time dependent version

**Comment by kke** [comment]
I suppose it could be done by specifying a [`--devel`](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#unstable-versions-devel-head) version in the Formula, but this means we would have to make a new PR to brew every time we release a pre-version.

**Comment by kke** [comment]
Of course the PR's could (and should) be automated.

**Comment by SpComb** [comment]
> tried that yesterday - it's difficult to tell from the --version of that if it's master or not (version says 1.4.0)

The CLI from the current master branch should have a `--version` of `1.4.0.dev`?

Given, the `master` branch `VERSION` should really be `1.4.1.dev` at this point (and hopefully `1.5.0.dev` soon).

**Comment by kke** [comment]
I think what we should do is:

* Setup a brew tap 
* Setup travis to update the formula and push it to the master branch of that tap and create a tag 
* Setup travis to strip the `devel` section of the formula when building a stable release and have it open a PR of that to homebrew-core
* Bonus: Set up something that updates the formula in tap from homebrew-core once the PR is merged to add the `bottle` section for latest stable (practically a list of tar.gz's for different OS versions of the installation directory, when available brew uses that to install instead of building from source).

With that, you can do:

```console
$ brew tap kontena/kontena
$ brew install kontena # would currently install 1.4.0
$ cd "$(brew --repo kontena/kontena)"
$ git checkout 1.4.1.rc1
$ brew install --devel kontena # installs 1.4.1.rc1
# toggle between versions:
$ brew switch kontena 1.4.0 
$ brew switch kontena 1.4.1.rc1
```

This way we don't have to make so many prerelease PR's to homebrew-core and people can install the latest stable immediately when it comes out instead of having to wait for the new version PR to be accepted. People that are not in such a rush can just install whatever comes from the homebrew-core or use `--HEAD` in either case to install the current master branch.
